File: gdx-gltf-starter/core/src/com/mygdx/game/terrains/attributes/TerrainFloatAttribute.java
package com.mygdx.game.terrains.attributes;

import com.badlogic.gdx.math.MathUtils;
import com.badlogic.gdx.utils.NumberUtils;

public class TerrainFloatAttribute extends TerrainAttribute {
    public static final String MinSlopeAlias = "minSlope";
    public static final long MinSlope = register(MinSlopeAlias);

    public static TerrainFloatAttribute createMinSlope(float value) {
        return new TerrainFloatAttribute(MinSlope, value);
    }

    public float value;

    public TerrainFloatAttribute(long type, float value) {
        super(type);
        this.value = value;
    }

    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 977 * result + NumberUtils.floatToRawIntBits(value);
        return result;
    }

    @Override
    public int compareTo(TerrainAttribute o) {
        if (type != o.type) return (int) (type - o.type);
        final float v = ((TerrainFloatAttribute) o).value;
        return MathUtils.isEqual(value, v) ? 0 : value < v ? -1 : 1;
    }
}

File: GDX_GLTF_Example/core/src/com/mygdx/game/terrains/attributes/TerrainFloatAttribute.kt
package com.mygdx.game.terrains.attributes

import com.badlogic.gdx.math.MathUtils
import com.badlogic.gdx.utils.NumberUtils

class TerrainFloatAttribute(type: Long, @JvmField var value: Float) : TerrainAttribute(type) {
    override fun hashCode(): Int {
        var result = super.hashCode()
        result = 977 * result + NumberUtils.floatToRawIntBits(value)
        return result
    }

    override fun compareTo(o: TerrainAttribute?): Int {
        if (type != o!!.type) return (type - o.type).toInt()
        val v = (o as TerrainFloatAttribute).value
        return if (MathUtils.isEqual(value, v)) 0 else if (value < v) -1 else 1
    }

    companion object {
        const val MinSlopeAlias: String = "minSlope"

        @JvmField
        val MinSlope: Long = register(MinSlopeAlias)

        @JvmStatic
        fun createMinSlope(value: Float): TerrainFloatAttribute {
            return TerrainFloatAttribute(MinSlope, value)
        }
    }
}
